docs/essentials/exceptions/exceptions-hiearchy.md

```diff
@@ -0,0 +1,29 @@
+# Exception hierarchy
+
+Exceptions can be generated by an executing program or the common language runtime, or they can be thrown by an application or the runtime.
+
+[Exception](http://dotnet.github.io/api/System.Exception.html) is the base class for all exceptions in .NET Core. Most exceptions that derive directly from **Exception** add no functionality to the **Exception** class. For example, the [InvalidCastException](http://dotnet.github.io/api/System.InvalidCastException.html) class hierarchy is as follows:
+
+```
+Object
+  Exception
+    SystemException
+       InvalidCastException
+```
+
+Severe exceptions, such as [OutOfMemoryException](http://dotnet.github.io/api/System.OutOfMemoryException.html), are thrown by the runtime or in unrecoverable conditions.
```

Comment:
I believe that these are actually called "critical exceptions".

I don't think we have a concept of "unrecoverable conditions", per se. There are a defined set of uncatchable exceptions. They are uncatchable due to the environment being unreliable. Unrecoverable is a stronger statement.